src/gamecontrol.cpp

```diff
@@ -15,13 +15,128 @@
 #include "people.h"
 #include "window.h"
 #include "dates.h"
-#include "gamemode.h"
 #include "viewport.h"
 #include "weather.h"
 #include "freerct.h"
 
+GameModeManager _game_mode_mgr; ///< Game mode manager object.
+
+/** Runs various procedures that have to be done yearly. */
+void OnNewYear()
+{
+	// Nothing (yet) needed.
+}
+
+/** Runs various procedures that have to be done monthly. */
+void OnNewMonth()
+{
+	_finances_manager.AdvanceMonth();
+	_rides_manager.OnNewMonth();
+}
+
+/** Runs various procedures that have to be done daily. */
+void OnNewDay()
+{
+	_rides_manager.OnNewDay();
+	_guests.OnNewDay();
+	_weather.OnNewDay();
+	NotifyChange(WC_BOTTOM_TOOLBAR, ALL_WINDOWS_OF_TYPE, CHG_DISPLAY_OLD, 0);
+}
+
+/**
+ * For every frame do...
+ * @param frame_delay Number of milliseconds between two frames.
+*/
+void OnNewFrame(uint32 frame_delay)
+{
+	_window_manager.Tick();
+	_guests.DoTick();
+	DateOnTick();
+	_guests.OnAnimate(frame_delay);
+	_rides_manager.OnAnimate(frame_delay);
+}
+
+GameControl::GameControl()
+{
+	this->running = true;
+}
+
+GameControl::~GameControl()
+{
+}
+
+/**
+ * Run latest game control action.
+ * @pre next_action should not be equal to #GCA_NONE.
+ */
+void GameControl::RunAction()
+{
+	assert(this->next_action != GCA_NONE);
```

Comment:
This forces the call pattern if (_game_control.next_action != GCA_NONE) _game_control.RunAction();

It's nicer if you don't force these details on the caller. Instead, add a wrapper method (inline in the .h file for example), where you handle this. The caller can just call stupidly _game_control.DoIt();, and the c++ compiler will inline the test and skip the call.